XTrophy.cs

```diff
@@ -92,6 +96,77 @@ public Int64 MyBalance()
             } else Error.Throw("XTrophyError: Operation denied");
         }
 
+
+        //// Withdrawal methods
+
+        // Freeze XTrophy.
+        public void WithdrawalRequest(Int64 amount, String hashCardNumber)
+        {
+            Require(amount > 0, "Amount must be positive");
+
+            Bytes sender = Info.Sender();
+            WithdrawalData withdrawalData = FrozenBalance.GetOrDefault(sender, new WithdrawalData());
+            Require(withdrawalData.amount == 0, "Withdrawal is already there");
+
+            Int64 balance = Balance.GetOrDefault(sender, 0);
+            Require(balance >= amount, "Not enough XTrophy for Withdrawal");
```

Comment:
You mean `balance < amount`?

Reply:
`Balance` is 100, `amount` is 50
100 >= 50 is true -> Not enough XTrophy for Withdrawal

Reply:
This a function Require. 
Require 100 >= 50 else error